compiler/rustc_data_structures/Cargo.toml

```diff
@@ -24,6 +24,7 @@ rayon-core = { version = "0.3.0", package = "rustc-rayon-core" }
 rustc-hash = "1.1.0"
 smallvec = { version = "1.0", features = ["union", "may_dangle"] }
 rustc_index = { path = "../rustc_index", package = "rustc_index" }
+rustc_ty_library = { path = "../rustc_ty_library" }
```

Comment:
This dependency will cause problems when moving more elaborate data structures to `rustc_ty_library`. Having `rustc_ty_library` depend on `rustc_data_structures` may be more robust.

Reply:
Eek. `rustc_data_structures` has a lot of depdencies, most of which we probably don't need/want for the eventual type library. Wondering if it's worth thinking about splitting `rustc_data_structures` or just giving in to the extra dependencies.

Granted, this sort of thing was something we expected to encounter rather quickly when starting this process, so it's not surprising at all.